File: 250320/특정 구간의 원소 평균값/elemental-mean-value-for-a-particular-interval.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const arr = input[1].trim().split(' ').map(Number);

// Please Write your code here.
let count = 0

for(let i = 0; i < n; i++){
    for(let j = i; j < n; j++){
        const temp = arr.slice(i, i=== j ? i + 1 : j + 2)
        const avg = temp.reduce((prev, cur) => prev + cur, 0) / temp.length
        if (avg % 1 === 0 && temp.includes(avg)) {
            count++;
        }
    }
}
console.log(count)

File: 250808/특정 구간의 원소 평균값/elemental-mean-value-for-a-particular-interval.py
n = int(input())
arr = list(map(int, input().split()))

# Please write your code here.
cnt = 0
for i in range(n):
    for j in range(i, n):
        temp = sum(arr[i:j+1])

        exist = False
        avg = temp / (j + 1 - i)
        for k in range(i, j+1):
            if(arr[k] == avg):
                exist = True
        
        if exist:
            cnt +=1

print(cnt)